### PR TITLE
8.0 sale prodlot mrp

### PR DESCRIPTION
**Pull request by Viggor**
[WIP] Add and transmit lot in mrp from sale. Work with https://github.com/OCA/sale-workflow/pull/177

### CODE REVIEW

**Comment by Viggor** [comment]
Will be adapted to work with https://github.com/OCA/sale-workflow/pull/140

**Comment by pedrobaeza** [comment]
I think this one is not covered, but you can make it a new module depending on the other one with this functionality

**Comment by Viggor** [comment]
Yes i will review https://github.com/OCA/sale-workflow/pull/140 and adapt my module to it

**Comment by pedrobaeza** [comment]
Thanks for collaborating!

File: sale_prodlot_mrp/mrp.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    """ Purpose to generate manufacturing base on custom product raw material
+    """
+    _inherit = 'mrp.production'
+    _service_product_lst = []
+    _product_config_dict = {}
+
+    lot_id = fields.Many2one('stock.production.lot', 'Lot')
+
+    @api.multi
+    def _action_compute_lines(self, properties=None):
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/mrp.py]: Please add a comment in the code, something like
We passe the production.id in the context so you can easily get retrieve the production information during the process that generate the line of needed for the production order. For exemple, you can in the function bom_explod read the information of the production order to customise the needed

File: sale_prodlot_mrp/mrp.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+##############################################################################
+
+from openerp import models, api, fields
+
+
+class MrpProduction(models.Model):
+    """ Purpose to generate manufacturing base on custom product raw material
+    """
+    _inherit = 'mrp.production'
+    _service_product_lst = []
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/mrp.py]: remove this two dead attribute "_service_product_lst" and "_product_config_dict"

File: sale_prodlot_mrp/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    'name': 'Production From Configuration',
+    'version': '1.0',
+    'category': 'Generic Modules',
+    'description': """
+        production from configuration
+    """,
+    'author': 'Akretion',
+    'website': '',
+    'depends': [
+        'sale_embedded_configuration',
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/__openerp__.py]: Fix the dependency, as based module have changed

File: sale_prodlot_mrp/stock.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    def _prepare_explode_move(self, cr, uid, move, line, context=None):
+        product_obj = self.pool['product.product']
+        lot_obj = self.pool['stock.production.lot']
+        if context is None:
+            context = {}
+        res = super(StockMove, self).\
+            _prepare_explode_move(cr, uid, move, line, context=context)
+        if context.get('explode_lot'):
+            if move.product_id.track_from_sale:
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/stock.py]: Be carefull if you rename the flag into "sale_prodlot_generation" 
https://github.com/OCA/sale-workflow/pull/179/files#r33909242

File: sale_prodlot_mrp/stock.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    def _prepare_lot_for_move(
+            self, cr, uid, product_id, move, lot_index, context=None):
+        lot_number = "%s-%03d" % (
+            move.restrict_lot_id.name, lot_index)
+        return {
+            'name': lot_number,
+            'product_id': product_id,
+            'company_id': move.company_id.id,
+            'config': move.restrict_lot_id.config,
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/stock.py]: The "config" field is not used anymore, remove it

File: sale_prodlot_mrp/tests/test_production_from_configuration.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                product['product_qty'], partner_id=self.partner_id
+            )['value']
+            order_line['order_id'] = self.sale_order_id
+            order_line['product_id'] = product_id
+            order_line['configuration'] = {}
+            self.order_line_obj.create(cr, uid, order_line)
+        self.sale_order_obj.action_button_confirm(cr, uid, [self.sale_order_id])
+
+
+class TestSuccess(BaseTest):
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/tests/test_production_from_configuration.py]: We should refactor and update the test

File: sale_prodlot_mrp/stock.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+#  see licence in __openerp__.py or http://www.gnu.org/licenses/agpl-3.0.txt
+#  Copyright (C) 2014 Akretion (http://www.akretion.com).
+#  @author David BEAL <david.beal@akretion.com>
+#
+##############################################################################
+
+from openerp.osv import orm
+
+
+class StockMove(orm.Model):
```
Comment (sebastienbeau) [inline sale_prodlot_mrp/stock.py]: In fact we should explode this module in 2 modules.
The first module is sale_order_lot_mrp (that depend on sale_order_lot_selection and mrp)
This module add the lot into the production order and also pass the production.id into the context

The second module should be sale_order_lot_generator_mrp (that depend on sale_order_lot_mrp and sale_order_lot_generator). This module will generate automatically a prodlot for the component of the BOM with the option "phantom" that have the option "track_from_sale". An real exemple is when you sold a customisable product that will be delivered in two part. Then your product have a phamton bom and the option "track_from_sale" and when we explode it we have the two part of the production with an automatic prodlot number generated

File: sale_prodlot_mrp/mrp.py (hunk 1 above)
Comment (Viggor) [inline sale_prodlot_mrp/mrp.py]: Done!

File: sale_prodlot_mrp/mrp.py (hunk 2 above)
Comment (Viggor) [inline sale_prodlot_mrp/mrp.py]: Done!

File: sale_prodlot_mrp/__openerp__.py (hunk 3 above)
Comment (Viggor) [inline sale_prodlot_mrp/__openerp__.py]: Done!

File: sale_prodlot_mrp/stock.py (hunk 4 above)
Comment (Viggor) [inline sale_prodlot_mrp/stock.py]: Done!

File: sale_prodlot_mrp/stock.py (hunk 5 above)
Comment (Viggor) [inline sale_prodlot_mrp/stock.py]: Done!

File: sale_prodlot_mrp/stock.py (hunk 7 above)
Comment (Viggor) [inline sale_prodlot_mrp/stock.py]: Done!

**Comment by pedrobaeza** [comment]
This is still red, so I'm closing. Feel free to reopen if still needed.

**Pull request closed without merging**